### ISSUE TITLE
"Failure in incremental resolver" from Analysis Server

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#27173]
Apologies for the truncation, we don't do full logging by default and I hit this randomly while using Dart Code. I'm using the 1.19 release version of the SDK.

I'll try and repro this, but it happened when I was working on some Dart to parse the API spec. The specific code is this revision:

https://github.com/Dart-Code/analysis-server-typescript-generator/tree/985ab71673429d91d8d75df8cd667edca65498ef

I started typing a comment as the first line inside `buildClasses` when the error occurred. I've attached a more detailed log of the traffic sent to/from the AS.

[Analysis Server log](https://github.com/dart-lang/sdk/files/440594/log.txt)

```shell
{
	"event": "server.error",
	"params": {
		"isFatal": false,
		"message": "Failure in incremental resolver",
		"stackTrace": "The null object does not have a getter 'iterator'.
		
		NoSuchMethodError: method not found: 'iterator'
		Receiver: null
		Arguments: []
		#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)
		#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)
		#2      IncrementalResolver._shiftErrors (package:analyzer/src/generated/incremental_resolver.dart:395)
		#3      IncrementalResolver._shiftErrors_NEW (package:analyzer/src/generated/incremental_resolver.dart:407)
		#4      IncrementalResolver._shiftEntryErrors (package:analyzer/src/generated/incremental_resolver.dart:388)
		#5      PoorMansIncrementalResolver.resolve (package:analyzer/src/generated/incremental_resolver.dart:593)
		#6      AnalysisContextImpl._tryPoorMansIncrementalResolution.<anonymous closure> (package:analyzer/src/context/context.dart:1988)
		#7      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:212)
		#8      AnalysisContextImpl._tryPoorMansIncrementalResolution (package:analyzer/src/context/context.dart:1951)
		#9      AnalysisContextImpl.handleContentsChanged (package:analyzer/src/context/context.dart:1108)
		#10     AnalysisServer.updateContent.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:1355)
		#11     List.forEach (dart:core-patch/growable_array.dart:258)
		#12     AnalysisServer.updateContent.<anonymous closure> (package:analysis_server/src/analysis_server.dart:1348)
		#13     _HashVMBase&MapMixin&&_LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:345)
		#14     AnalysisServer.updateContent (package:analysis_server/src/analysis_server.dart:1293)
		#15     AnalysisDomainHandler.updateContent (package:analysis_server/src/domain_analysis.dart:302)
		#16     AnalysisDomainHandler.handleRequest (package:analysis_server/src/domain_analysis.dart:208)
		#17     AnalysisServer.h...
```

**Comment by DanTup** [comment]
Unfortunately I'm unable to repro this; hopefully the stack trace might give clues about what might've happened?

**Comment by DanTup** [comment]
Actually, I can reproduce this relatively reliably. Open the file `api_spec.dart` from the mentioned repo, but the cursor at the end of the line that starts `void buildClasses(` and then press enter twice quickly.

Sometimes it doesn't work, but then pressing the delete key does. It's possible this is timing-related, though I've been unable to repro in a simple sample file.

**Comment by bwilkerson** [comment]
@scheglov

**Comment by scheglov** [comment]
Fixed in e95eb9e794ddb607d67e499d41f5f962a811669e already.
